src/pip/_internal/req/constructors.py

```diff
@@ -239,7 +236,7 @@ def install_req_from_editable(
         install_options=options.get("install_options", []) if options else [],
         global_options=options.get("global_options", []) if options else [],
         hash_options=options.get("hashes", {}) if options else {},
-        extras=parts.extras,
+        extras=parts.extras if parts.extras else set(),
```

Comment:
Can we change this in `parse_req_from_line` instead?

Reply:
Can I change the type here from `Optional[Set[str]` to  `Set[str]` and return `set()` instead of `None`. That's the only place we are using `Optional[Set[str]` 

https://github.com/pypa/pip/blob/76b865155ee646927e838c377b46643b51ccd2fd/src/pip/_internal/req/constructors.py#L80-L82

If not, what exactly do we change in `parse_req_from_line` ?

Reply:
Yea, that sounds good to me!

Reply:
Oops, this caused some tests to fail which were asserting to None (https://dev.azure.com/pypa/pip/_build/results?buildId=24265&view=logs&j=88d9dc86-3d79-5e8a-7a2d-b92d46a16d40&t=9a9e1fc6-59d9-5186-707a-ec9179d5f4bd&l=61). Do I update those tests to go with the update in `parse_editable` as well?

Reply:
Yup! Sounds good to me!

Reply:
I suggest splitting out the `parse_editable`-related changes into a separate commit, and having another commit for adding/updating all the other type annotations.

Reply:
So I have https://github.com/pypa/pip/pull/8292/commits/60121b272eb0837da4625164741b48bff910298b for the `parse_editable` changes to source, and https://github.com/pypa/pip/pull/8292/commits/8abd988b5da748499af04e1149795ef24d31824a for the related changes in tests. Do I combine these in one?

Reply:
nah, we're good. :)